### PR TITLE
promise nothing

### PR DESCRIPTION
**Pull request by kuhe**
\* all reviewers welcome

Separate PR, because I feel like fetch and promise are important to some people.

This
```js
        fetch(path)
            .then(handleData.bind(this))
            .then(handleSuccess.bind(this))
            .catch(handleError.bind(this));
```

could be this:

```js
this.xhr(path, handleError, handleData, handleSuccess);
...
[request.responseText, ...yep].reduce((a, b) => b(a));
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,16 +50,14 @@
...
-    "build-js-all": "npm run build-js-vendor && npm run build-js && npm run build-js-full",
+    "build-js": "browserify src/scripts/g5-component-browser.js --s 'g5-component' | uglifyjs --mangle --compress drop_console,drop_debugger,dead_code,unused > dist/g5-component.min.js",
+    "build-js-dev": "browserify src/scripts/g5-component-browser.js --s 'g5-component' > dist/g5-component.js",
+    "build-js-all": "npm run build-js && npm run build-js",
     "build-css": "lessc --include-path=node_modules/bootstrap/less:src/styles src/styles/component.less > dist/g5-component.css",
     "postbuild-css": "npm run minify-css && npm run gzip-css",
     "build": "npm run build-js-all && npm run build-css && npm run compress-images",
     "prebuild": "echo 'Running all builds...'",
-    "postbuild": "echo 'Builds are ready!'",
+    "postbuild": "npm run du; echo 'Builds are ready!'",
```
Comment (gbabula) [inline package.json]: `du`?

File: package.json [hunk 2]
```diff
@@ -38,8 +38,7 @@
       ]
     ]
   },
-  "browser": {
-  },
+  "browser": {},
```
Comment (gbabula) [inline package.json]: We can kill this property if we don't need it.

File: src/scripts/utils/master.js [hunk 3]
```diff
@@ -26,22 +27,34 @@ const utils = {
...
-    log() {
+    log(...args) {
 
         const timestamp = utils.timestamp;
-        const args = Array.prototype.slice.call(arguments);
 
         args.unshift(`${timestamp()} - g5-component :`);
 
+        /**
+         * If you want stack traces in your logs, enable below.
```
Comment (gbabula) [inline src/scripts/utils/master.js]: Would be nice to have this in.

File: src/scripts/model/master.js [hunk 4]
```diff
@@ -149,6 +125,38 @@ class MasterModel extends EventEmitter {
     }
 
     /**
+     * @param {string} url data location.
+     * @param {Function} uhOh <*(Error)> an error handler.
+     * @param {...Function} yep any number of success handlers.
+     */
+    xhr(url, uhOh, ...yep) {
```
Comment (gbabula) [inline src/scripts/model/master.js]: lets add a real variable name for the error case

**Comment by gbabula** [comment]
I'm a fan of `fetch` _(as you know)_, mainly because its standardized.

The `Promise` polyfill is only included in this project because of the `fetch` dependency. 
I can see your argument here and I actually don't mind this change since its abstracted away from the developer.

File: package.json (hunk 1 above)
Comment (kuhe) [inline package.json]: I could make the script called "disk-usage"? The command line program is "du".

File: src/scripts/utils/master.js (hunk 3 above)
Comment (kuhe) [inline src/scripts/utils/master.js]: Turns out to be a lot of output. Gotta keep it clean for Web Media to use :p

Comment (kuhe) [inline src/scripts/utils/master.js]: screen shot 2017-04-12 at 10 05 58 am

How about a second method "trace(...args)"?

File: package.json (hunk 1 above)
Comment (gbabula) [inline package.json]: Gotcha, `disk-usage` sounds good, more obvious at least.

File: src/scripts/utils/master.js (hunk 3 above)
Comment (gbabula) [inline src/scripts/utils/master.js]: Second method sounds good, or hide it under a flag.

**Comment by melissalabbe** [comment]
What is the benefit of moving from fetch to xhr? Is it mainly to avoid the extra dependencies needed for cross-browser support? Ultimately I'm ok with the change, if a developer wants to override the master model they can still choose to include fetch in their custom model, but it does feel like a step away from standards.

File: src/scripts/utils/master.js (hunk 3 above)
Comment (kuhe) [inline src/scripts/utils/master.js]: What kind of flag?

**Comment by kuhe** [comment]
@melissalabbe 

The current benefit is a reduction of delivered code, es6-promise and what-wg-fetch. For a polling usage such as this, XHR accomplishes the same thing in a few lines, and is equally standard.

**Pull request merged**